### PR TITLE
[flang] Update printing values in dump-parse-tree

### PR DESCRIPTION
**Pull request by kparzysz**
Remove 'if std::string' that is covered by another branch of the if-statement.
Add printing of 'bool' and 'int' values, since they have corresponding `GetNodeName` definitions.

### CODE REVIEW

**Comment by kiranchandramohan** [comment]
Can we have a test since this non-NFC?

**Comment by kparzysz** [comment]
There aren't really any tests for dumping the parse tree, except for OpenMP.  The difference (at least for the `int` case) would show in a declaration like `real, allocatable :: z(:)`.  I can add an OpenMP testcase that has this declaration plus some OMP construct for the sake of being an OpenMP test...  What do you think?

**Comment by kparzysz** [comment]
Source:
```
subroutine foo()
  real, allocatable :: z(:)
end
```

Tree before:
```
Program -> ProgramUnit -> SubroutineSubprogram
| SubroutineStmt
| | Name = 'foo'
| SpecificationPart
| | ImplicitPart ->
| | DeclarationConstruct -> SpecificationConstruct -> TypeDeclarationStmt
| | | DeclarationTypeSpec -> IntrinsicTypeSpec -> Real
| | | AttrSpec -> Allocatable
| | | EntityDecl
| | | | Name = 'z'
| | | | ArraySpec -> DeferredShapeSpecList -> int
| ExecutionPart -> Block
| EndSubroutineStmt ->
```

Tree after (compare the 3rd line from the bottom):
```
Program -> ProgramUnit -> SubroutineSubprogram
| SubroutineStmt
| | Name = 'foo'
| SpecificationPart
| | ImplicitPart ->
| | DeclarationConstruct -> SpecificationConstruct -> TypeDeclarationStmt
| | | DeclarationTypeSpec -> IntrinsicTypeSpec -> Real
| | | AttrSpec -> Allocatable
| | | EntityDecl
| | | | Name = 'z'
| | | | ArraySpec -> DeferredShapeSpecList -> int = '1'
| ExecutionPart -> Block
| EndSubroutineStmt ->
```

File: flang/include/flang/Parser/dump-parse-tree.h [hunk 1]
```diff
@@ -884,8 +884,10 @@ class ParseTreeDumper {
     } else if constexpr (HasSource<T>::value) {
       return x.source.ToString();
 #endif
-    } else if constexpr (std::is_same_v<T, std::string>) {
```
Comment (klausler) [inline flang/include/flang/Parser/dump-parse-tree.h:887]: are you sure that you want to delete this case?  is it redundant or something?

Comment (kparzysz) [reply flang/include/flang/Parser/dump-parse-tree.h:887]: Yes, it's already covered at line 874.

**Review by klausler — APPROVED** [review]

**Review by kiranchandramohan — APPROVED** [review]
> I can add an OpenMP testcase that has this declaration plus some OMP construct for the sake of being an OpenMP test... What do you think?

If you can add a test that would be great. But no pressure.

**Pull request merged**